Submitted By: Joe Ciccone <jciccone@gmail.com>
Date: 14-06-2010
Initial Package Version: 2.6.18
Upstream Status: Unknown
Origin: Joe Ciccone
Description: This patch gives the ability to change the libdir.
             Updated for Iproute2-2.6.28 by Jim Gifford
             Updated for Iproute2-2.6.34 by Joe Ciccone
             Updated for Iproute2-2.6.37 by Joe Ciccone
             Renamed for Iproute2-2.6.38 by Jonathan Norman
             Rediffed for Iproute2-3.4.0 by William Harrington

diff -Naur iproute2-3.4.0.orig/Makefile iproute2-3.4.0/Makefile
--- iproute2-3.4.0.orig/Makefile	2012-05-21 21:12:19.000000000 +0000
+++ iproute2-3.4.0/Makefile	2012-08-26 18:34:17.338322481 +0000
@@ -13,7 +13,7 @@
 
 SHARED_LIBS = y
 
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\" -DARPDDIR=\"$(ARPDDIR)\"
 ifneq ($(SHARED_LIBS),y)
 DEFINES+= -DNO_SHARED_LIBS
 endif
diff -Naur iproute2-3.4.0.orig/Makefile.orig iproute2-3.4.0/Makefile.orig
--- iproute2-3.4.0.orig/Makefile.orig	1970-01-01 00:00:00.000000000 +0000
+++ iproute2-3.4.0/Makefile.orig	2012-05-21 21:12:19.000000000 +0000
@@ -0,0 +1,85 @@
+ROOTDIR=$(DESTDIR)
+PREFIX=/usr
+LIBDIR=$(PREFIX)/lib
+SBINDIR=/sbin
+CONFDIR=/etc/iproute2
+DATADIR=$(PREFIX)/share
+DOCDIR=$(DATADIR)/doc/iproute2
+MANDIR=$(DATADIR)/man
+ARPDDIR=/var/lib/arpd
+
+# Path to db_185.h include
+DBM_INCLUDE:=$(ROOTDIR)/usr/include
+
+SHARED_LIBS = y
+
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+ifneq ($(SHARED_LIBS),y)
+DEFINES+= -DNO_SHARED_LIBS
+endif
+
+DEFINES+=-DCONFDIR=\"$(CONFDIR)\"
+
+#options if you have a bind>=4.9.4 libresolv (or, maybe, glibc)
+LDLIBS=-lresolv
+ADDLIB=
+
+#options for decnet
+ADDLIB+=dnet_ntop.o dnet_pton.o
+
+#options for ipx
+ADDLIB+=ipx_ntop.o ipx_pton.o
+
+CC = gcc
+HOSTCC = gcc
+DEFINES += -D_GNU_SOURCE
+CCOPTS = -O2
+WFLAGS = -Wall -Wstrict-prototypes
+CFLAGS = $(WFLAGS) $(CCOPTS) -I../include $(DEFINES)
+YACCFLAGS = -d -t -v
+
+SUBDIRS=lib ip tc misc netem genl man
+
+LIBNETLINK=../lib/libnetlink.a ../lib/libutil.a
+LDLIBS += $(LIBNETLINK)
+
+all: Config
+	@set -e; \
+	for i in $(SUBDIRS); \
+	do $(MAKE) $(MFLAGS) -C $$i; done
+
+Config:
+	sh configure $(KERNEL_INCLUDE)
+
+install: all
+	install -m 0755 -d $(DESTDIR)$(SBINDIR)
+	install -m 0755 -d $(DESTDIR)$(CONFDIR)
+	install -m 0755 -d $(DESTDIR)$(ARPDDIR)
+	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples
+	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples/diffserv
+	install -m 0644 README.iproute2+tc $(shell find examples -maxdepth 1 -type f) \
+		$(DESTDIR)$(DOCDIR)/examples
+	install -m 0644 $(shell find examples/diffserv -maxdepth 1 -type f) \
+		$(DESTDIR)$(DOCDIR)/examples/diffserv
+	@for i in $(SUBDIRS) doc; do $(MAKE) -C $$i install; done
+	install -m 0644 $(shell find etc/iproute2 -maxdepth 1 -type f) $(DESTDIR)$(CONFDIR)
+
+snapshot:
+	echo "static const char SNAPSHOT[] = \""`date +%y%m%d`"\";" \
+		> include/SNAPSHOT.h
+
+clean:
+	@for i in $(SUBDIRS) doc; \
+	do $(MAKE) $(MFLAGS) -C $$i clean; done
+
+clobber:
+	touch Config
+	$(MAKE) $(MFLAGS) clean
+	rm -f Config cscope.*
+
+distclean: clobber
+
+cscope:
+	cscope -b -q -R -Iinclude -sip -slib -smisc -snetem -stc
+
+.EXPORT_ALL_VARIABLES:
diff -Naur iproute2-3.4.0.orig/include/iptables.h iproute2-3.4.0/include/iptables.h
--- iproute2-3.4.0.orig/include/iptables.h	2012-05-21 21:12:19.000000000 +0000
+++ iproute2-3.4.0/include/iptables.h	2012-08-26 18:34:17.338322481 +0000
@@ -4,8 +4,12 @@
 #include "iptables_common.h"
 #include "libiptc/libiptc.h"
 
+#ifndef LIBDIR
+#define LIBDIR "/usr/lib"
+#endif
+
 #ifndef IPT_LIB_DIR
-#define IPT_LIB_DIR "/usr/local/lib/iptables"
+#define IPT_LIB_DIR LIBDIR "/iptables"
 #endif
 
 #ifndef IPPROTO_SCTP
diff -Naur iproute2-3.4.0.orig/misc/arpd.c iproute2-3.4.0/misc/arpd.c
--- iproute2-3.4.0.orig/misc/arpd.c	2012-05-21 21:12:19.000000000 +0000
+++ iproute2-3.4.0/misc/arpd.c	2012-08-26 18:34:17.338322481 +0000
@@ -40,7 +40,7 @@
 int resolve_hosts;
 
 DB	*dbase;
-char	*dbname = "/var/lib/arpd/arpd.db";
+char	*dbname = ARPDDIR "/arpd.db";
 
 int	ifnum;
 int	*ifvec;
